core/src/main/java/org/apache/iceberg/deletes/SortingPositionOnlyDeleteWriter.java

```diff
@@ -41,14 +41,19 @@
  * records are not ordered by file and position as required by the spec. If the incoming deletes are
  * ordered by an external process, use {@link PositionDeleteWriter} instead.
  *
+ * <p>If configured, this writer can also load previous deletes using the provided function and
+ * merge them with incoming ones prior to writing. Only file-scoped deletes are added to the list of
+ * rewritten delete files that can be discarded upon the commit.
+ *
  * <p>Note this writer stores only positions. It does not store deleted records.
  */
 public class SortingPositionOnlyDeleteWriter<T>
     implements FileWriter<PositionDelete<T>, DeleteWriteResult> {
 
   private final Supplier<FileWriter<PositionDelete<T>, DeleteWriteResult>> writers;
   private final DeleteGranularity granularity;
-  private final CharSequenceMap<Roaring64Bitmap> positionsByPath;
+  private final CharSequenceMap<PositionDeleteIndex> positionsByPath;
```

Comment:
Is this an area we'd want to explore using Map<String, PositionDeleteIndex> instead of the `CharSequenceMap`? Doesn't need to be in this PR, more so just wondering

Reply:
Actually, I think it'll probably make more sense to look at that when I do the update to use `location` instead of the deprecated `path`.

Reply:
We may want to keep this as `CharSequenceMap` as writers may use arbitrary `CharSequence` implementations and it is a bit different from `DataFile`/`DeleteFile` structs.